.github/workflows/gradle.yml

```diff
@@ -9,25 +9,30 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      checks: write
+      contents: read
     strategy:
       matrix:
         java: ['11']
     env:
       WORKSPACE: ${{ github.workspace }}
       GRADLE_OPTS: -Xmx1500m -Dfile.encoding=UTF-8
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v4
         with:
           distribution: 'adopt'
           java-version: ${{ matrix.java }}
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          # Publish a Gradle build scan so we can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of the build.
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
       - name: Run Tests
```

Comment:
When looking over all your comments on the Conversation tab of this pull request, I missed this comment. 

I was actually going to ask you about this. I was under the assumption that adding these environment variables to this step is unnecessary because the Gradle `clean` command deletes any cached dependencies anyways... 
```
env:
  DEVELOCITY_BUILD_CACHE_NODE_USER: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER }}
  DEVELOCITY_BUILD_CACHE_NODE_KEY: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY }}
```
But I'm honestly not sure if adding them would slow down execution of the workflow or speed it up.

Reply:
Here's the results of my test runs of the Java CI workflow _without_ the build cache env variables:
[50 seconds for "Run Tests", 1m 6s total for the workflow](https://github.com/tylervz/grails-static-website/actions/runs/11844527351/job/33007953078)
[48 seconds for "Run Tests", 59 seconds total for the workflow](https://github.com/tylervz/grails-static-website/actions/runs/11844609509/job/33008232832)

Here's my test runs of the Java CI workflow _with_ the build cache env variables:
[48 seconds for "Run Tests", 1m 0s total for the workflow](https://github.com/tylervz/grails-static-website/actions/runs/11844930158/job/33009267590)
[48 seconds for "Run Tests", 1m 0s total for the workflow](https://github.com/tylervz/grails-static-website/actions/runs/11844977266/job/33009417171)
[46 seconds for "Run Tests", 56 seconds total for the workflow](https://github.com/tylervz/grails-static-website/actions/runs/11845003870/job/33009504367)

Because adding the build cache environment variables seemed to have no negative impact on execution speed, I will add them for the sake of consistency.